.github/workflows/tests.yml

```diff
@@ -0,0 +1,209 @@
+        with:
+          repository: MANTRA-Chain/cosmos-sdk
+          path: cosmos-sdk
+          ref: ${{ env.SDK_VERSION }}
+      - name: Install tparse
+        run: |
+          go install github.com/mfridman/tparse@latest
+      - name: Run all tests
+        run: |
+          # Define directories to skip
+          SKIP_DIRS=(
+            "tools"
+            "orm"
+            # Add more directories here as needed
+          )
+          for dir in "${SKIP_DIRS[@]}"; do
+            if [ -d "$dir" ]; then
+              echo "Skipping tests for $dir"
+              rm -rf "$dir"
+            fi
+          done
+          
+          make test-all
+        working-directory: cosmos-sdk
+
+  test-e2e:
+    name: E2E Tests
+    runs-on: ubuntu-latest-x64-xlarge
+    steps:
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.23
+      - uses: actions/checkout@v4
+      - name: Install tparse
+        run: |
+          go install github.com/mfridman/tparse@latest
+      - run: make test-e2e
+
+  test-e2e-nix:
+    name: E2E Nix Tests
+    runs-on: ubuntu-latest-x64-xlarge
+    timeout-minutes: 240
+    strategy:
+      matrix:
+        tests: [all]
+    env:
+      TESTS_TO_RUN: ${{ matrix.tests }}
+    steps:
+      - name: Create more disk space
+        run: sudo rm -rf /usr/share/dotnet && sudo rm -rf /opt/ghc && sudo rm -rf "/usr/local/share/boost" && sudo rm -rf "$AGENT_TOOLSDIRECTORY"
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - id: changed-files
+        uses: tj-actions/changed-files@ed68ef82c095e0d48ec87eccea555d944a631a4c # v46.0.5
+        with:
+          files: |
+            docs
+            *.md
+            **/*.md
+      - uses: actions/setup-go@v5
+        with:
+          go-version: 1.23
+        if: steps.changed-files.outputs.only_changed == 'false'
+      - name: install
+        run: make install
+        if: steps.changed-files.outputs.only_changed == 'false'
+      - uses: cachix/install-nix-action@6a9a9e84a173d90b3ffb42c5ddaf9ea033fad011 # v23
+        with:
+          nix_path: nixpkgs=channel:nixos-25.05
+          extra_nix_config: |
+            access-tokens = github.com=${{ secrets.GITHUB_TOKEN }}
+        if: steps.changed-files.outputs.only_changed == 'false'
+      - uses: cachix/cachix-action@6a9a34cdd93d0ae4b4b59fd678660efb08109f2f # v12
+        if: steps.changed-files.outputs.only_changed == 'false'
+        with:
+          name: mantrachain-e2e
+          extraPullNames: dapp
+          # github don't pass secrets for pull request from fork repos,
+          # in that case the push is disabled naturally.
+          authToken: "${{ secrets.CACHIX_AUTH_TOKEN }}"
+
+      - name: Checkout mantrachain-e2e
+        if: steps.changed-files.outputs.only_changed == 'false'
+        uses: actions/checkout@v4
+        with:
+          repository: MANTRA-Chain/mantrachain-e2e
+          path: mantrachain-e2e
+          ref: main
```

Comment:
we map e2e [release/v5](https://github.com/MANTRA-Chain/mantrachain-e2e/tree/release/v5) to [release/v5.0.0-rcx](https://github.com/MANTRA-Chain/mantrachain/tree/release/v5.0.0-rcx), [main](https://github.com/MANTRA-Chain/mantrachain-e2e/tree/main) to [main](https://github.com/MANTRA-Chain/mantrachain/tree/main) now